File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayOrderRef.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@AllArgsConstructor @NoArgsConstructor
public class MallpayOrderRef  extends ApiBase implements Signable{
    private MallpayPrice totalPrice;
    private List<MallpayVat> totalVat;
    private List<MallpayOrderItemRef> items;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        if(totalPrice != null) add(sb, totalPrice.toSign());
        if(totalVat != null) totalVat.forEach(vat -> add(sb, vat.toSign()));
        if(items != null) for(MallpayOrderItemRef item: items) add(sb, item.toSign());
        deleteLast(sb);
        return sb.toString();
    }
}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayOrderRef.cs
using System.Text;
using CsobGatewayClientExample.Communication.DataObjects.Ext;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayOrderRef : ApiBase
{
    [JsonProperty("totalPrice")] public MallpayPrice? TotalPrice { get; set; }

    [JsonProperty("totalVat")] public List<MallpayVat>? TotalVat { get; set; }

    [JsonProperty("items")] public List<MallpayOrderItemRef>? Items { get; set; }

    [JsonProperty("extensions")] public List<Extension>? Extensions { get; set; }


    public string ToSign()
    {
        var sb = new StringBuilder();
        if (TotalPrice != null) Add(sb, TotalPrice?.ToSign());
        if (TotalVat != null)
            foreach (var vat in TotalVat)
                Add(sb, vat.ToSign());
        if (Items != null)
            foreach (var item in Items)
                Add(sb, item.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}